dias/analyzers/flag_rfi_analyzer.py

```diff
@@ -0,0 +1,592 @@
+                                    _fraction_flagged(mask_missing[:, ss, :],
+                                                      logical_not=True))
+
+                        self.fraction_masked_before.labels(stack=lbl).set(
+                                    _fraction_flagged(mask_before[:, ss, :],
+                                                      logical_not=True))
+
+                        self.fraction_masked_after.labels(stack=lbl).set(
+                                    _fraction_flagged(mask_after[:, ss, :],
+                                                      logical_not=True))
+
+                    # Garbage collect
+                    del data
+                    gc.collect()
+
+        # Increment run counter.
+        self.run_counter.inc()
+        self.run_timer.set(int(time.time() - self.run_start_time))
+
+    def update_data_index(self, start, stop, filename=None):
+        """Add row to data index database.
+
+        Update the data index database with a row that
+        contains the name of the file and the span of time
+        the file contains.
+
+        Parameters
+        ----------
+        start : unix time
+            Earliest time contained in the file.
+        stop : unix time
+            Latest time contanied in the file.
+        filename : str
+            Name of the file.
+        """
+        # Parse arguments
+        dt_start = ephemeris.unix_to_datetime(ephemeris.ensure_unix(start))
+        dt_stop = ephemeris.unix_to_datetime(ephemeris.ensure_unix(stop))
+
+        relpath = None
+        if filename is not None:
+            relpath = os.path.relpath(filename, self.write_dir)
+
+        # Insert row for this file
+        try:
+            cursor = self.data_index.cursor()
+            cursor.execute("INSERT INTO files VALUES (?, ?, ?)",
+                           (dt_start, dt_stop, relpath))
+
+        except Exception as ex:
+            self.log.error("Could not perform database insert: %s" % ex)
+
+        else:
+            self.data_index.commit()
+
+    def refresh_data_index(self):
+        """Remove expired rows from the data index database.
+
+        Remove any rows of the data index database
+        that correspond to files that have been cleaned
+        (removed) by dias manager.
+        """
+        cursor = self.data_index.cursor()
+        query = 'SELECT filename FROM files ORDER BY start'
+        all_files = list(cursor.execute(query))
+
+        for result in all_files:
+
+            filename = result[0]
+
+            if not os.path.isfile(os.path.join(self.write_dir, filename)):
+
+                try:
+                    cursor = self.data_index.cursor()
+                    query = 'DELETE FROM files WHERE filename = ?', (filename,)
+                    cursor.execute(query)
+
+                except Exception as ex:
+                    self.log.error("Could not perform database deletion: %s" %
```

Comment:
Same here as above: Is this supposed to be an error-log with no further consequences?